make command fails with a conversion error from ceres

##### System information (version)
- OpenCV => 4.x
- Operating System / Platform => Ubuntu 18.04 x86_64
- Compiler => gcc 9.4.0
- CUDA version => 11.3
- CUDNN Version => 8.4.1

##### Detailed description

I have a ubuntu system with 1050 GTX Mobile GPU. I am trying to build opencv for sfm and 3d recon from source, and have ceres, eigen installed from source as well. Both of them seem to be working fine. When I tried to build opencv from source I keep getting this error and the build stops abruptly during make -j6.

```shell
opencv_contrib/modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc: In function ‘void libmv::EuclideanBundleCommonIntrinsics(const libmv::Tracks&, int, int, libmv::EuclideanReconstruction*, libmv::CameraIntrinsics*, libmv::BundleEvaluation*)’:
/data/softwares/opencv_build/opencv_contrib/modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc:553:37: error: cannot convert ‘ceres::SubsetManifold*’ to ‘ceres::LocalParameterization*’
  553 |                                     constant_translation_manifold);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                     |
      |                                     ceres::SubsetManifold*
In file included from /usr/local/include/ceres/ceres.h:66,
                 from /data/softwares/opencv_build/opencv_contrib/modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc:25:
/usr/local/include/ceres/problem.h:396:51: note:   initializing argument 2 of ‘void ceres::Problem::SetParameterization(double*, ceres::LocalParameterization*)’
  396 |                            LocalParameterization* local_parameterization);
      |                            ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
modules/sfm/src/libmv/libmv/simple_pipeline/CMakeFiles/simple_pipeline.dir/build.make:62: recipe for target 'modules/sfm/src/libmv/libmv/simple_pipeline/CMakeFiles/simple_pipeline.dir/bundle.cc.o' failed
```
##### Steps to reproduce

The cmake command,
```shell
cmake -D CMAKE_BUILD_TYPE=RELEASE     -D CMAKE_INSTALL_PREFIX=/usr/local     -D INSTALL_C_EXAMPLES=ON     -D INSTALL_PYTHON_EXAMPLES=ON     -D OPENCV_GENERATE_PKGCONFIG=ON     -D OPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules     -D BUILD_EXAMPLES=ON         -D WITH_CUDA=ON     -D ENABLE_FAST_MATH=1     -D CUDA_FAST_MATH=1     -D WITH_CUBLAS=1 ..
```

Followed by
```shell
make -j6
```

And the error above shows up.

##### Issue submission checklist

 - [ ✅ ] I report the issue, it's not a question
 - [ ✅ ] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found any solution
 - [ ✅ ] I updated to the latest OpenCV version and the issue is still there 
 - [ ✅ ] There is reproducer code and related data files: videos, images, onnx, etc

--- a/modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc
+++ b/modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc
@@ -549,7 +549,7 @@ void EuclideanBundleCommonIntrinsics(
 
       if (bundle_constraints & BUNDLE_NO_TRANSLATION) {
 #if CERES_VERSION_MAJOR >= 3 || (CERES_VERSION_MAJOR >= 2 && CERES_VERSION_MINOR >= 1)
-        problem.SetParameterization(current_camera_R_t,
+        problem.SetManifold(current_camera_R_t,
                                     constant_translation_manifold);
 #else
         problem.SetParameterization(current_camera_R_t,

